tests/testthat/test-vivax.R

```diff
@@ -24,3 +24,53 @@ test_that('Test difference between falciparum and vivax parameter lists', {
   expect_true(all(names(vivax_parameters[!names(vivax_parameters) %in% names(falciparum_parameters)]) %in%
                     c("du_max","du_min","ku","au50","b","phi0lm","phi1lm","ic0lm","kclm","d0","ca","init_idm","f","gammal")))
 })
+
+## Test subpatent progression functions
+test_that('Test subpatent duration function works ', {
+  parameters <- get_parameters(overrides = list(s_proportion = 0,
+                                                d_proportion = 0,
+                                                a_proportion = 0,
+                                                u_proportion = 1,
+                                                t_proportion = 0))
+  vivax_parameters <- get_parameters(parasite = "vivax",
+                                     overrides = list(s_proportion = 0,
+                                                      d_proportion = 0,
+                                                      a_proportion = 0,
+                                                      u_proportion = 1,
+                                                      t_proportion = 0))
+  variables <- create_variables(vivax_parameters)
+  index <- variables$state$get_index_of("U")
+  expect_equal(object = subpatent_duration(parameters = vivax_parameters, variables = variables, index = index), 
+               expected = rep(vivax_parameters$du_max,100))
+  
+  ## Change initial values of ID, and IDM, check they are the same
+  variables$id <- individual::DoubleVariable$new(1:100)
+  ID_durations <- subpatent_duration(parameters = vivax_parameters, variables = variables, index = index)
+  
+  variables$id <- individual::DoubleVariable$new(rep(0,100))
+  variables$idm <- individual::DoubleVariable$new(1:100)
+  IDM_durations <- subpatent_duration(parameters = vivax_parameters, variables = variables, index = index)
+  
+  expect_equal(object = ID_durations, expected = IDM_durations)
+  
+  ## Check convergence to min_du at high immunity
+  variables$id <- individual::DoubleVariable$new(rep(1E7,100))
+  variables$idm <- individual::DoubleVariable$new(rep(0,100))
+  expect_equal(object = subpatent_duration(parameters = vivax_parameters, variables = variables, index = index),
+               expected = rep(vivax_parameters$du_min,100),
+               tolerance = 1E-2)
+})
+
+test_that('Test idm rendered creates output', {
```

Comment:
We should be able to test this without the model run.
Check out some of Giovanni's examples in tests/testthat/test-render.R

Reply:
I've had a go with this (ripped off Giovanni's tests, really) - see what you think!